src/spikeinterface/preprocessing/deepinterpolation/generators.py

```diff
@@ -0,0 +1,311 @@
+from __future__ import annotations
+import tempfile
+import json
+from typing import Optional
+import numpy as np
+
+from ...core import load_extractor, concatenate_recordings, BaseRecording, BaseRecordingSegment
+
+from .tf_utils import has_tf, import_tf
+
+from ...core import load_extractor
+
+from deepinterpolation.generator_collection import SequentialGenerator
+
+
+class SpikeInterfaceRecordingGenerator(SequentialGenerator):
+    """
+    This generator is used when dealing with a SpikeInterface recording.
+    The desired shape controls the reshaping of the input data before convolutions.
+    """
+
+    def __init__(
+        self,
+        recordings: BaseRecording | list[BaseRecording],
+        pre_frame: int = 30,
+        post_frame: int = 30,
+        pre_post_omission: int = 1,
+        desired_shape: tuple = (192, 2),
+        batch_size: int = 100,
+        steps_per_epoch: int = 10,
+        start_frame: Optional[int] = None,
+        end_frame: Optional[int] = None,
+        total_samples: int = -1,
+    ):
+        if not isinstance(recordings, list):
+            recordings = [recordings]
+        self.recordings = recordings
+        if len(recordings) > 1:
+            assert (
+                r.get_num_channels() == recordings[0].get_num_channels() for r in recordings[1:]
+            ), "All recordings must have the same number of channels"
+
+        total_num_samples = np.sum(r.get_total_samples() for r in recordings)
+        # In case of multiple recordings and/or multiple segments, we calculate the frame periods to be excluded (borders)
+        exclude_intervals = []
+        pre_extended = pre_frame + pre_post_omission
```

Comment:
I think the SequentialGenerator might have some of this logic already?

Reply:
to exclude borders? I couldn't find it!